python/ray/serve/deployment_graph.py

```diff
@@ -40,7 +40,7 @@ def _deserialize(cls, *args):
     def __reduce__(self):
         return RayServeDAGHandle._deserialize, (self.dag_node_json,)
 
-    async def remote(self, *args, **kwargs) -> ray.ObjectRef:
+    async def remote(self, *args, _cache_refs: bool = False, **kwargs) -> ray.ObjectRef:
```

Comment:
can this be passed in through any other method? the `*args` and `**kwargs**` here are supposed to take whatever users are sending, we should not inject our own value here.
- We should at least prefix it with `_ray`.
- Maybe consider a new private method like `_remote_with_cache_refs`
- Maybe env var?